pytealutils/inline/inline_asm.py

```diff
@@ -0,0 +1,48 @@
+from pyteal import CompileOptions, Expr, LeafExpr, Mode, TealBlock, TealOp, TealType
+
+
+class CustomOp:
+    def __init__(self, opcode):
+        self.opcode = opcode
+        self.mode = Mode.Signature | Mode.Application
+        self.min_version = 2
+
+    def __str__(self) -> str:
+        return self.opcode
+
+
+class InlineAssembly(LeafExpr):
+    """InlineAssembly can be used to inject TEAL source directly into a PyTEAL program
+
+        Often the generated PyTEAL is not as efficient as it can be. This class offers a way to
+        write the most efficient TEAL for code paths that would otherwise be impossible because
+        of opcode budget constraints.
+
+        It can also be used to implement methods that are not yet available in the PyTEAL repository
+
+        Args
+            opcode: string containing the teal to inject
+            args: any number of PyTEAL expressions to place before this opcode
+            type: The type this Expression returns, to help during PyTEAL compilation
+
+    """
+
+    def __init__(
+        self, opcode: str, *args: "Expr", type: TealType = TealType.none
+    ) -> None:
+        super().__init__()
+        opcode_with_args = opcode.split(" ")
```

Comment:
```suggestion
        opcode_with_args = opcode.split()
```

Compare for example ouputs of `splitme.split()` v. `splitme.split(" ")` to see why I'm recommending where:
```python
splitme = "hello world  double   triple   "
```

Reply:
I implemented this but backed it out for now, the resultant teal pushed all the separate lines into the same line which is not what I'm looking for